sos/cleaner/__init__.py

```diff
@@ -0,0 +1,554 @@
+                    checksum = self.generate_checksum(arc.final_archive_path)
+                    if checksum is not None:
+                        dname = "checksums/%s.%s" % (arc_dest, self.hash_name)
+                        self.archive.add_string(checksum, dest=dname)
+                else:
+                    for dirname, dirs, files in os.walk(arc.archive_path):
+                        for filename in files:
+                            if filename.startswith('sosreport'):
+                                continue
+                            fname = os.path.join(dirname, filename)
+                            dnm = fname.split(arc.archive_name)[-1].lstrip('/')
+                            self.archive.add_file(fname, dest=dnm)
+            arc_path = self.archive.finalize(self.opts.compression_type)
+        else:
+            arc = self.completed_reports[0]
+            arc_path = arc.final_archive_path
+            checksum = self.generate_checksum(arc.final_archive_path)
+            if checksum is not None:
+                chksum_name = "%s.%s" % (arc_path.split('/')[-1],
+                                         self.hash_name)
+                with open(os.path.join(self.sys_tmp, chksum_name), 'w') as cf:
+                    cf.write(checksum)
+
+        final_path = os.path.join(self.sys_tmp, arc_path.split('/')[-1])
+        shutil.move(arc_path, final_path)
+        arcstat = os.stat(final_path)
+
+        # logging will have been shutdown at this point
+        print("A mapping of obfuscated elements is available at\n\t%s"
+              % map_path)
+
+        print("\nThe obfuscated archive is available at\n\t%s\n" % final_path)
+        print("\tSize\t%s" % get_human_readable(arcstat.st_size))
+        print("\tOwner\t%s\n" % getpwuid(arcstat.st_uid).pw_name)
+
+        print("Please send the obfuscated archive to your support "
+              "representative and keep the mapping file private")
+
+        self.cleanup()
+
+    def compile_mapping_dict(self):
+        """Build a dict that contains each parser's map as a key, with the
+        contents as that key's value. This will then be written to disk in the
+        same directory as the obfuscated report so that sysadmins have a way
+        to 'decode' the obfuscation locally
+        """
+        _map = {}
+        for parser in self.parsers:
+            _map[parser.map_file_key] = {}
+            _map[parser.map_file_key].update(parser.mapping.dataset)
+
+        return _map
+
+    def write_map_to_file(self, _map, path):
+        """Write the mapping to a file on disk that is in the same location as
+        the final archive(s).
+        """
+        with open(path, 'w') as mf:
+            mf.write(json.dumps(_map, indent=4))
+        os.chmod(path, 0o600)
+        return path
+
+    def write_map_for_archive(self, _map):
+        try:
+            map_path = os.path.join(self.sys_tmp,
+                                    "private_map_%s" % self.arc_name)
```

Comment:
A subjective suggestion: could the `private_map` be appended to the end of the filename, instead of beginning? Similarly like `.md5` checksum? Pros: it will be easier to remove old sosreports (`rm -rf /var/tmp/sos*`) as well as one more easily identifies the files as related to a sosreport. Negs: maybe a customer might make a mistake and upload `sosreport-pmoravec-rhel8-2020-06-02-trfaglz.tar.gz` altogether with `sosreport-pmoravec-rhel8-2020-06-02-trfaglz.tar.gz_private_map`.

Reply:
Ack